File: examples/TreeList/AllProjects.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from omero.rtypes import rstring
from omero_sys_ParametersI import ParametersI


def getProjects(query_prx, username):
    return query_prx.findAllByQuery(
        "select p from Project p"
        " join fetch p.datasetLinks dil"
        " join fetch dil.child"
        " where p.details.owner.omeName = :name",
        ParametersI().add("name", rstring(username)))

File: examples/TreeList/AllProjects.m
function projects = AllProjects(query, username)
q = ['select p from Project p join fetch p.datasetLinks dil ',...
     'join fetch dil.child where p.details.owner.omeName = :name'];
p = omero.sys.ParametersI();
p.add('name', omero.rtypes.rstring(username));
projects = query.findAllByQuery(q, p);
